cool-core/src/main/java/com/nus/cool/core/cohort/refactor/CohortProcessor.java

```diff
@@ -127,12 +128,25 @@ private void processCublet(CubletRS cublet) {
      * @param chunk
      * @param hashMapperBySchema
      */
-    private void processDataChunk(ChunkRS chunk, HashMap<String, String[]> hashMapperBySchema) {
+    private void processDataChunk(ChunkRS chunk, HashMap<String, String[]> hashMapperBySchema, UserMetaFieldRS userMetaField) {
+        List<InputVector> userToInvariant=userMetaField.getUserToInvariant();
         for (int i = 0; i < chunk.getRecords(); i++) {
             // load data into tuple
             for (String schema : this.projectedSchemaSet) {
                 // if the value is segment type, we should convert it to String from globalId
-                if(hashMapperBySchema.containsKey(schema)){
+                if(chunk.getSchema().getInvariantName2Id().containsKey(schema)){
+                    String idName=chunk.getSchema().getField(chunk.getSchema().getUserKeyField()).getName();
```

Comment:
create a method with meaningful name to encapsulate this logic for better readability